### PR TITLE
Do not show aces as as birdie in par 2 holes

### PR DESCRIPTION
**Pull request by kajte**
This fixes bug in score distribution chart. Hole in ones are not showed as birdies in par 2 holes and as pars in par 1 holes.

### CODE REVIEW

File: app/models/Course.php [hunk 1]
```diff
@@ -443,8 +443,8 @@ public static function score_distribution($courseid) {
                 SELECT hole.hole_num,
                 CASE WHEN score.stroke = 1 THEN 1 END AS hole_in_one,
                 CASE WHEN hole.par - score.stroke - score.ob = 2 AND score.stroke != 1 THEN 1 END AS eagle,
-                CASE WHEN hole.par - score.stroke - score.ob = 1 THEN 1 END AS birdie,
-                CASE WHEN hole.par - score.stroke - score.ob = 0 THEN 1 END AS par,
+                CASE WHEN hole.par - score.stroke - score.ob = 1 AND score.stroke != 1 THEN 1 END AS birdie,
+                CASE WHEN hole.par - score.stroke - score.ob = 0 AND score.stroke != 1 THEN 1 END AS par,
```
Comment (virtalas) [inline app/models/Course.php]: The first case should already catch it when stroke = 1?

**Comment by virtalas** [comment]
If there ever is such a (stupid) thing as a par 1 hole, I don't think it should count as a hole in one. Could add `AND hole.par != 1` to the first case. But that's such an edge case it's not really necessary.

File: app/models/Course.php (hunk 1 above)
Comment (kajte) [reply app/models/Course.php]: It should'nt catch it because each of them are separate cases. One case doesn't affect to other cases.

Comment (virtalas) [reply app/models/Course.php]: Oh I see

Comment (virtalas) [reply app/models/Course.php]: Then rather remove this check for par and add it to the first case as I described in the other comment?

**Comment by kajte** [comment]
Okay, now we have a support for par 1 holes in score distribution. Hopefully that prevents ace hunter course designers from doing such a shame as a par 1 hole.

**Comment by virtalas** [comment]
This piece of code is a little prone to error. Need to be careful that same scores aren't counted twice as different things.

**Pull request merged**